### ISSUE TITLE
(Steam + IOS) Various texture issues

### ISSUE THREAD

**Issue by emeryhyden** [fortellgames/bw_bugtracker#240]
General inconsistency: about 40% of the textures apply flipped 90 degrees compared to the preview

Several textures are a warning symbol (⚠︎) and cannot be applied to any surface
![warning1](https://github.com/user-attachments/assets/ee331c34-ea5d-45cd-ab8c-f64fd7c006b5)
![warning2](https://github.com/user-attachments/assets/516e3f60-a6c2-41bb-b69e-2e002e07c274)

Some textures do not apply to blocks 
![cannotapply](https://github.com/user-attachments/assets/ef287776-2bb1-4bdd-a43c-9713015165cc)

Some textures only apply to the sides of a block and not the top surface
![cannotapplytotop](https://github.com/user-attachments/assets/ed16b511-2cb1-45e3-b0b2-dd5f450282a6)
![cannotapplytotop2](https://github.com/user-attachments/assets/596226b7-25fb-4d91-96d2-b20a3e37dc99)
(the rainbow texture has many issues and seems to be the most broken texture)

Some textures darken/brighten or otherwise change the color of other surfaces
![darkensthesides](https://github.com/user-attachments/assets/63d625fc-c25a-4c2e-9782-3f0254411fdc)

When placed on a very large surface, many of the transparent colors cannot be painted or do not paint correctly, and/or have a strange blue gradient no matter the color
![gradient](https://github.com/user-attachments/assets/97420531-a2d4-4e4e-872d-c22d874679ae)

Some textures unexpectedly repeat rather than being extended (an option to switch between stretch and repeat textures would be a very robust solution for all textures)
![repeating](https://github.com/user-attachments/assets/d7891980-a649-4d3a-98d9-48f4dc49683c)

**Comment by Aktion** [comment]
Awesome, thanks for that! Is it for both iOS and Steam?

**Comment by emeryhyden** [comment]
Yes, the issue is identical on both platforms. Once I get my paws on the VR/MR version I will check for the issues there

**Comment by cetfen** [comment]
Adding onto this, most of these bugs have existed since the linden builds of Blocksworld, although the transparent texture issues are most definitely new.
Textures applying 90 degrees compared to the preview might be because the block you're applying them to was rotated, or you're looking at it from a different angle.
The warning sign textures actually can be applied to surfaces. The first one makes blocks invisible, but doesn't hide reflections. The second acts similarly to the texture remover (the Ø with a grey background), but it can apply to terrain. If i had to guess, the reason why these textures are a warning symbol in the block pallet is because the warning symbol is a fallback texture, like the source purple checkerboard.

Regarding the issue of textures affecting the color of a block, I've seen people use this bug to expand the range of colors they can use. Perhaps it'd be best to add a color wheel (maybe only accessible after some tutorials) to satisfy those kinds of players.

**Comment by emeryhyden** [comment]
you're absolutely right on the rotation issue, i was just looking from the wrong orientation
my mistake, sorry for the trouble

**Comment by Aktion** [comment]
@cetfen  @emeryhyden i need this issue to cleaned as its too much text - need to convert to action points
